src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9328,19 +9346,17 @@ private bool ScanDesignator()
             }
         }
 
-        private bool IsPossibleLambdaExpression(Precedence precedence)
+        private bool IsPossibleAnonymousMethodExpression()
         {
-            if (precedence <= Precedence.Lambda && this.PeekToken(1).Kind == SyntaxKind.EqualsGreaterThanToken)
+            // Skip past any static/async keywords.
+            var tokenIndex = 0;
+            while (this.PeekToken(tokenIndex).Kind == SyntaxKind.StaticKeyword ||
```

Comment:
>this.PeekToken(tokenIndex) [](start = 19, length = 26)

Consider extracting a local.

Reply:
PeekToken tends ot be very fast (it either primes an array, or looks right into one).  And in practice we'll be looking at 0-1 items max.  I'd prefer to leave this as is.